src/clj/rems/home.clj

```diff
@@ -14,15 +15,22 @@
             [ring.util.response :refer [content-type not-found redirect response]])
   (:import [rems.auth UnauthorizedException]))
 
-(defn- apply-for-resource [resource]
+(defn- resource-to-item [resource]
   (let [items (->> (catalogue/get-localized-catalogue-items {:resource resource})
                    (filter :enabled))]
     (cond
-      (= 0 (count items)) (-> (not-found "Resource not found")
-                              (content-type "text/plain"))
-      (< 1 (count items)) (-> (not-found "Resource ID is not unique")
-                              (content-type "text/plain"))
-      :else (redirect (str "/application?items=" (:id (first items)))))))
+      (= 0 (count items)) :not-found
+      (< 1 (count items)) :not-unique
+      :else (:id (first items)))))
+
+(defn- apply-for-resources [resources]
+  (let [items (map resource-to-item resources)]
+    (cond
+      (some #{:not-found} items) (-> (not-found "Resource not found")
+                                     (content-type "text/plain"))
+      (some #{:not-unique} items) (-> (not-found "Resource ID is not unique")
```

Comment:
Possibly we could use the 400 bad request here?

Reply:
why not!